test/e2e/scripts/compare-shard-mappings.js

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env node
+
+const [,, nodeMappingJson, oldMappingJson, newMappingJson] = process.argv;
+
+if (!nodeMappingJson || !oldMappingJson || !newMappingJson) {
+  console.error('Usage: compare-shard-mappings.js <node-mapping-json> <old-mapping-json> <new-mapping-json>');
+  process.exit(1);
+  /* Usage:
+     node_mapping='{"couchdb@couchdb-1.local":"couchdb@couchdb-1-namespace-svc-cluster.local",
+                    "couchdb@couchdb-2.local":"couchdb@couchdb-2-namespace-svc-cluster.local",
+                    "couchdb@couchdb-3.local":"couchdb@couchdb-3-namespace-svc-cluster.local"}'
+     node ./scripts/compare-shard-mappings.js "$node_mapping" "$shard_mapping" "$shard_mapping2"
+  */
+}
+
+function compareMappings(nodeMappingJson, oldMappingJson, newMappingJson) {
+  const nodeMapping = JSON.parse(nodeMappingJson);
+  const oldMapping = JSON.parse(oldMappingJson);
+  const newMapping = JSON.parse(newMappingJson);
+
+  Object.keys(oldMapping).forEach(shardRange => {
+    if (!(shardRange in newMapping)) {
+      console.error(`Shard range ${shardRange} from original mapping not found in new mapping`);
+      process.exit(1);
+    }
+
+    const oldShardDbs = oldMapping[shardRange];
+    const newShardDbs = newMapping[shardRange];
+
+    Object.keys(oldShardDbs).forEach(dbName => {
+      if (!(dbName in newShardDbs)) {
+        console.error(`Database ${dbName} in shard range ${shardRange} not found in new mapping`);
+        process.exit(1);
+      }
+
+      const oldNode = oldShardDbs[dbName];
+      const newNode = newShardDbs[dbName];
+
+      // Map old node name to new node name using the nodeMapping
+      const expectedNewNode = nodeMapping[oldNode] || oldNode;
```

Comment:
I think adding this test check is great! 

This line easily make the test invalid, because if our code doesn't do anything - as in doesn't move any shard, always allowing the old node to be mapped will never fail this script. 

I think we should just be very rigid and strict about the input to this script and always pass exactly which nodes have to have been moved. 

```suggestion
const expectedNewNode = nodeMapping[oldNode];
```

Reply:
The `|| oldNode` can be useful if you prefer not to specify all nodes in the cluster but only those that you migrated.

As long as the nodeMapping contains a node that's supposed to have been changed, this test will fail if that migration hasn't happened.

However, to avoid confusion, I have removed it. That means we always have to specify all nodes to the compare script (whether or not they have been migrated)

Reply:
yes! that's exactly what I wanted. since this script is used to validate a test, maximum strictness is required.